problems/[230302] 101. Symmetric Tree.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/symmetric-tree
+ *
+ * Definition for a binary tree node.
+ * function TreeNode(val, left, right) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.left = (left===undefined ? null : left)
+ *     this.right = (right===undefined ? null : right)
+ * }
+ */
+/**
+ * @param {TreeNode} root
+ * @return {boolean}
+ */
+const isSymmetric = function (root) {
+  const leftTreeValues = [];
+
+  (function leftTreeDfs(tree) {
+    if (!tree) {
+      leftTreeValues.push(101);
+
+      return;
+    }
+
+    leftTreeValues.push(tree.val);
+
+    if (tree.left || tree.right) {
+      leftTreeDfs(tree.left);
+      leftTreeDfs(tree.right);
+    }
+  })(root.left);
+
+  function compareRightToTheLeftTreeItems(tree, LeftItemsArr) {
+    const compareLeftValue = LeftItemsArr.pop();
+
+    if (!tree) {
+      if (compareLeftValue !== 101) {
+        LeftItemsArr.push(101);
+      }
+
+      return;
+    }
+
+    if (compareLeftValue !== tree.val) {
+      LeftItemsArr.push(101);
+
+      return;
+    }
+
+    if (tree.left || tree.right) {
+      compareRightToTheLeftTreeItems(tree.right, LeftItemsArr);
+      compareRightToTheLeftTreeItems(tree.left, LeftItemsArr);
```

Comment:
```suggestion
      const rightResult = compareRightToTheLeftTreeItems(tree.right, LeftItemsArr);

if (rightResult === false) {
  return false;
}
      compareRightToTheLeftTreeItems(tree.left, LeftItemsArr);
```

위에서 101을 넣고 return시켜서 재귀를 종료시키는 방식으로 symmetric하지 않은 경우를 만들어주는군요..! 그런데 right에서 재귀가 종료되어도 left에 대한 연산을 또 진행할 수가 있을 것 같아요. 이 경우는 결과가 나왔음에도 불필요하게 연산이 되는 경우일텐데요. 별도로 return값을 주는 방식으로 한다면(제 예시는 false로 들었습니당) 따로 연산이 안이루어질 수 있지 않을까요!

Reply:
역쉬 재혁님 예리하시군요.. 👍